### ISSUE TITLE
Unable to connect to database (1-15) other than the default database 0 when using redis 

### ISSUE THREAD

**Issue by ziliwesley** [go-gitea/gitea#15680]
- Gitea version (or commit ref): 1.14.1 (using offical Docker image)
- Git version: Not Related
- Operating system: Not Related
 
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No

## Description

I have been trying to setup a gitea deployment with kubernetes on my own machines with an **external PostgreSQL** as persistence store and an **external Redis** as cache store (session store as well).

Here is part of my `/data/gitea/conf/app.ini`

```ini
[session]
COOKIE_SECURE   = true
PROVIDER        = redis
# I have tried using this before, same outcome.
# PROVIDER_CONFIG: network=tcp,addr=some.host.addr:6379,password=some_password,db=14,pool_size=100,idle_timeout=180
PROVIDER_CONFIG = redis://:some_password@some.host.addr:6379/14?pool_size=100&idle_timeout=100
```

After gitea sign in, session related key could be written to Redis successfully; the problem is, **but only to the default database: db0**; The database I chose (which is **db14**)  got nothing.

### Thoughts

After some reading of the source code, I found it maybe related to codes below

[modules/nosql/redis.go](https://github.com/go-gitea/gitea/blob/master/modules/nosql/redis.go#L78)

```go
...
	case "db":
		uri.Path = "/" + items[1]
...
```

and [modules/nosql/manager_redis.go](https://github.com/go-gitea/gitea/blob/master/modules/nosql/manager_redis.go#L188)

```go
...
	case "redis":
		if uri.Host != "" {
			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
		}
		if uri.Path != "" {
			if db, err := strconv.Atoi(uri.Path); err == nil {
				opts.DB = db
			}
		}
		client.UniversalClient = redis.NewClient(opts.Simple())
...
```

I also wrote some simple codes to test it:

```go
	ret, err := strconv.Atoi("/14")

	if err != nil {
		// Will panic: strconv.Atoi: parsing "/14": invalid syntax
		panic(err)
	}

	log.Printf("parsed: %v", ret)
```

**Comment by wULLSnpAXbWZGYDYyhWTKKspEQoaYxXyhoisqHf** [comment]
this string works fine for me, although it's the db0..
`network=tcp,addr=:6379,password=ultralongandrandom,db=0,pool_size=100,idle_timeout=180`

that's what's not working as well, right?

**Comment by ziliwesley** [comment]
> this string works fine for me, although it's the db0..
> `network=tcp,addr=:6379,password=ultralongandrandom,db=0,pool_size=100,idle_timeout=180`
> 
> that's what's not working as well, right?

Yes. It's working fine to connect the *db0* using both methods.

**Comment by zeripath** [comment]
Yup looks like you've spotted a bug! Should be fixed by #15698
